src/api/utils/__tests__/apiUtils.test.js

```diff
@@ -0,0 +1,152 @@
+import { HttpError } from 'react-admin';
+
+import { client } from '../../apolloClient/client';
+import {
+  queryHandler,
+  mutationHandler,
+  normalizeApiTranslations,
+  denormalizeLocalTranslations,
+  mapApiDataToLocalData,
+  mapLocalDataToApiData,
+  handleApiConnection,
+  handleApiNode,
+} from '../apiUtils';
+
+vi.mock('../apolloClient/client');
+
+describe('apiUtils', () => {
+  describe('normalizeApiTranslations', () => {
+    it('normalizes API translations to local format', () => {
+      const apiTranslations = [
+        {
+          languageCode: 'FI',
+          name: 'foo',
+          address: 'bar',
+          __typename: 'bazType',
+        },
+      ];
+      const expected = {
+        FI: { name: 'foo', address: 'bar' },
+      };
+
+      const result = normalizeApiTranslations(apiTranslations);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('denormalizeLocalTranslations', () => {
+    it('denormalizes local translations to API format', () => {
+      const localTranslations = {
+        FI: { name: 'foo', address: 'bar' },
+      };
+      const expected = [{ languageCode: 'FI', name: 'foo', address: 'bar' }];
+
+      const result = denormalizeLocalTranslations(localTranslations);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('mapApiDataToLocalData', () => {
+    it('maps API data to local data format', () => {
+      const apiNode = {
+        id: '1',
+        translations: [
+          {
+            languageCode: 'FI',
+            name: 'foo',
+            address: 'bar',
+            __typename: 'bazType',
+          },
+        ],
+        __typename: 'bazType',
+      };
+      const expected = {
+        id: '1',
+        translations: {
+          FI: { name: 'foo', address: 'bar' },
+        },
+      };
+
+      const result = mapApiDataToLocalData(apiNode);
+      expect(result).toEqual(expected);
+    });
+
+    it('returns null if apiNode is nullish', () => {
```

Comment:
```suggestion
    it('returns null if apiNode is null', () => {
```
As now it's only testing with null.